### PR TITLE
Print to PDF improvements

### PR DESCRIPTION
**Pull request by nirvn**
Print to PDF improvements:
- when a single layout is present, skip the submenu and simply print that one
- when printing, throw a message to the user
- delay the printing for some milliseconds to show UI feedback

The single layout improvement was a(nother) good suggestion from @nyalldawson.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 004eb910f828fad86d933891fefb6eb361600409:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-1b222ac47c188516713c510ecc54ff1966a3046a-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-1b222ac47c188516713c510ecc54ff1966a3046a-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-1b222ac47c188516713c510ecc54ff1966a3046a-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-1b222ac47c188516713c510ecc54ff1966a3046a-x86.apk)

**Comment by nirvn** [comment]
Print menu tip moved to sit _above_ the print layout items:
image

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 78ef82ec8112f847bc7d0837b0e424f216619bca:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-405d1f6dd45c5eb8c78b72cfe1a26db5dd333632-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-405d1f6dd45c5eb8c78b72cfe1a26db5dd333632-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-405d1f6dd45c5eb8c78b72cfe1a26db5dd333632-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-405d1f6dd45c5eb8c78b72cfe1a26db5dd333632-x86.apk)

File: src/qml/qgismobileapp.qml [hunk 1]
```diff
@@ -1135,8 +1135,22 @@ ApplicationWindow {
...
+          printMenu.popup( mainMenu.x, mainMenu.y + printItem.y )
+          highlighted = false
+        }
+        else
+        {
+          mainMenu.close();
+          displayToast( qsTr( 'Printing to PDF') )
+          printMenu.printTimer.interval = 350;
+          printMenu.printTimer.repeat = false;
+          printMenu.printTimer.triggered.connect(function () {
```
Comment (signedav) [inline src/qml/qgismobileapp.qml:1149]: Just made a quick test on mobile. On selecting "print to pdf" the first time it sends the open intent one time, on selecting it the second time, it opens two times, on selecting it the third time, three times and so on. Could it be that it conects the signal multiple times?

Comment (nirvn) [reply src/qml/qgismobileapp.qml]: @signedav , fixed.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 3ab34d1a93ae883570651188fbd3bfd83df05990:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-c310c5e9c8f01da1a4954878d6fa71e161a6c78b-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-c310c5e9c8f01da1a4954878d6fa71e161a6c78b-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-c310c5e9c8f01da1a4954878d6fa71e161a6c78b-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-c310c5e9c8f01da1a4954878d6fa71e161a6c78b-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for b676023f916bda685c1f7365d357eb2c627f3ef8:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-c49b6ccebddbf1f87b4af90ccffbd1ce8744938a-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-c49b6ccebddbf1f87b4af90ccffbd1ce8744938a-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-c49b6ccebddbf1f87b4af90ccffbd1ce8744938a-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-c49b6ccebddbf1f87b4af90ccffbd1ce8744938a-x86.apk)

File: src/qml/qgismobileapp.qml [hunk 2]
```diff
@@ -1135,8 +1135,18 @@ ApplicationWindow {
...
+          printMenu.popup( mainMenu.x, mainMenu.y + printItem.y )
+          highlighted = false
+        }
+        else
+        {
+          mainMenu.close();
+          displayToast( qsTr( 'Printing to PDF') )
+          printMenu.printIndex = 0
+          printMenu.printTimer.restart();
+        }
```
Comment (signedav) [inline src/qml/qgismobileapp.qml:1149]:
```suggestion
        }
        else
        {
          mainMenu.close();
          displayToast( qsTr( 'Printing to PDF') )
          printMenu.printIndex = 0
          printMenu.printTimer.restart();
        }
        highlighted = false
```
I think the highlighted needs to be false in every case.

**Review by signedav — APPROVED** [review]
Besides the highlight it looks good. Thanks.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 965ea781f9e4fc935d5c74332e16dee1fa9153d5:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-5e7b24cd662a4f10f4b5009c44ec850686cb2558-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-5e7b24cd662a4f10f4b5009c44ec850686cb2558-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-5e7b24cd662a4f10f4b5009c44ec850686cb2558-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1535-5e7b24cd662a4f10f4b5009c44ec850686cb2558-x86.apk)

**Pull request merged**